packages/std/src/results/system_result.rs

```diff
@@ -0,0 +1,76 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+use super::super::errors::SystemError;
+
+/// This is the outer result type returned by a querier to the contract.
+///
+/// We use a custom type here instead of Rust's Result because we want to be able to
+/// define the serialization, which is a public interface. Every language that compiles
+/// to Wasm and runs in the ComsWasm VM needs to create the same JSON representation.
+///
+/// # Examples
+///
+/// Success:
+///
+/// ```
+/// # use cosmwasm_std::{to_vec, Binary, ContractResult, SystemResult};
+/// let data = Binary::from(b"hello, world");
+/// let result = SystemResult::Ok(ContractResult::Ok(data));
+/// assert_eq!(to_vec(&result).unwrap(), br#"{"ok":{"ok":"aGVsbG8sIHdvcmxk"}}"#.to_vec());
+/// ```
+///
+/// Failure:
+///
+/// ```
+/// # use cosmwasm_std::{to_vec, Binary, ContractResult, SystemResult, SystemError};
+/// let error = SystemError::Unknown {};
+/// let result: SystemResult<Binary> = SystemResult::Err(error);
+/// assert_eq!(to_vec(&result).unwrap(), br#"{"error":{"unknown":{}}}"#.to_vec());
+/// ```
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum SystemResult<S> {
```

Comment:
This is for consistent serde serialization, right?

`"error": ...` rather than `"Err": ...`

Reply:
Right, I want to have full control over the serialization